@ipa-fmw 
Travis does not have DepthSenseSDK so that it cannot successfully build up.
Installing DepthSenseSDK requires membership login, therefore I think it is not good idea to add travis for this repo.

Travis log is as below.

```
# create empty overlay workspace

mkdir -p $CATKIN_WS_SRC

SCRIPT IS RUNNING

source $CATKIN_WS_UNDERLAY/install/setup.bash > /dev/null 2>&1 # source install space of underlay

catkin_init_workspace $CATKIN_WS_SRC

Creating symlink "/home/travis/catkin_ws/src/CMakeLists.txt" pointing to "/opt/ros/indigo/share/catkin/cmake/toplevel.cmake"

cd $CATKIN_WS

catkin_make -DCMAKE_BUILD_TYPE=Release # build empty overlay

Base path: /home/travis/catkin_ws

Source space: /home/travis/catkin_ws/src

Build space: /home/travis/catkin_ws/build

Devel space: /home/travis/catkin_ws/devel

Install space: /home/travis/catkin_ws/install

####

#### Running command: "cmake /home/travis/catkin_ws/src -DCMAKE_BUILD_TYPE=Release -DCATKIN_DEVEL_PREFIX=/home/travis/catkin_ws/devel -DCMAKE_INSTALL_PREFIX=/home/travis/catkin_ws/install -G Unix Makefiles" in "/home/travis/catkin_ws/build"

####

-- The C compiler identification is GNU 4.8.4

-- The CXX compiler identification is GNU 4.8.4

-- Check for working C compiler: /usr/bin/cc

-- Check for working C compiler: /usr/bin/cc -- works

-- Detecting C compiler ABI info

-- Detecting C compiler ABI info - done

-- Detecting C compile features

-- Detecting C compile features - done

-- Check for working CXX compiler: /usr/bin/c++

-- Check for working CXX compiler: /usr/bin/c++ -- works

-- Detecting CXX compiler ABI info

-- Detecting CXX compiler ABI info - done

-- Detecting CXX compile features

-- Detecting CXX compile features - done

-- Using CATKIN_DEVEL_PREFIX: /home/travis/catkin_ws/devel

-- Using CMAKE_PREFIX_PATH: /home/travis/catkin_ws_underlay/install;/opt/ros/indigo

-- This workspace overlays: /home/travis/catkin_ws_underlay/install;/opt/ros/indigo

-- Found PythonInterp: /usr/bin/python (found version "2.7.6") 

-- Using PYTHON_EXECUTABLE: /usr/bin/python

-- Using Debian Python package layout

-- Using empy: /usr/bin/empy

-- Using CATKIN_ENABLE_TESTING: ON

-- Call enable_testing()

-- Using CATKIN_TEST_RESULTS_DIR: /home/travis/catkin_ws/build/test_results

-- Looking for include file pthread.h

-- Looking for include file pthread.h - found

-- Looking for pthread_create

-- Looking for pthread_create - not found

-- Looking for pthread_create in pthreads

-- Looking for pthread_create in pthreads - not found

-- Looking for pthread_create in pthread

-- Looking for pthread_create in pthread - found

-- Found Threads: TRUE  

-- Found gtest sources under '/usr/src/gtest': gtests will be built

-- Using Python nosetests: /usr/bin/nosetests-2.7

-- catkin 0.6.16

-- BUILD_SHARED_LIBS is on

-- Configuring done

-- Generating done

-- Build files have been written to: /home/travis/catkin_ws/build

####

#### Running command: "make -j1" in "/home/travis/catkin_ws/build"

####

# populate overlay

ln -s $TRAVIS_BUILD_DIR $CATKIN_WS_SRC

# install dependencies from overlay

rosdep install -q --from-paths $CATKIN_WS_SRC -i -y --rosdistro $CI_ROS_DISTRO > /dev/null #2>&1

Extracting templates from packages: 100%

Not building database; man-db/auto-update is not 'true'.

Not building database; man-db/auto-update is not 'true'.

Extracting templates from packages: 100%

Not building database; man-db/auto-update is not 'true'.

# build overlay

source $CATKIN_WS/devel/setup.bash > /dev/null 2>&1 # source devel space of overlay

catkin_make -DCMAKE_BUILD_TYPE=Release

Base path: /home/travis/catkin_ws

Source space: /home/travis/catkin_ws/src

Build space: /home/travis/catkin_ws/build

Devel space: /home/travis/catkin_ws/devel

Install space: /home/travis/catkin_ws/install

####

#### Running command: "cmake /home/travis/catkin_ws/src -DCMAKE_BUILD_TYPE=Release -DCATKIN_DEVEL_PREFIX=/home/travis/catkin_ws/devel -DCMAKE_INSTALL_PREFIX=/home/travis/catkin_ws/install -G Unix Makefiles" in "/home/travis/catkin_ws/build"

####

-- Using CATKIN_DEVEL_PREFIX: /home/travis/catkin_ws/devel

-- Using CMAKE_PREFIX_PATH: /home/travis/catkin_ws/devel;/home/travis/catkin_ws_underlay/install;/opt/ros/indigo

-- This workspace overlays: /home/travis/catkin_ws/devel;/home/travis/catkin_ws_underlay/install;/opt/ros/indigo

-- Using PYTHON_EXECUTABLE: /usr/bin/python

-- Using Debian Python package layout

-- Using empy: /usr/bin/empy

-- Using CATKIN_ENABLE_TESTING: ON

-- Call enable_testing()

-- Using CATKIN_TEST_RESULTS_DIR: /home/travis/catkin_ws/build/test_results

-- Found gtest sources under '/usr/src/gtest': gtests will be built

-- Using Python nosetests: /usr/bin/nosetests-2.7

-- catkin 0.6.16

-- BUILD_SHARED_LIBS is on

-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-- ~~  traversing 2 packages in topological order:

-- ~~  - softkinetic (metapackage)

-- ~~  - softkinetic_camera

-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-- +++ processing catkin metapackage: 'softkinetic'

-- ==> add_subdirectory(softkinetic/softkinetic)

-- +++ processing catkin package: 'softkinetic_camera'

-- ==> add_subdirectory(softkinetic/softkinetic_camera)

-- Using these message generators: gencpp;genlisp;genpy

-- Could NOT find DepthSenseSDK (missing:  DepthSenseSDK_LIBRARY DepthSenseSDK_INCLUDE_DIR) 

CMake Error at softkinetic/softkinetic_camera/CMakeLists.txt:20 (message):

  +++++++++ Could not locate the DepthSense SDK +++++++++

  Please install the SDK before trying to build the 'softkinetic_camera'

  package, see README.txt.



  Cannot continue, aborting.
```

--- a/softkinetic_camera/cfg/Softkinetic.cfg
+++ b/softkinetic_camera/cfg/Softkinetic.cfg
@@ -32,6 +32,9 @@ gen.add("vfov", double_t, 0, "Vertical FOV (degrees)", 90.0, 0.0, 180.0)
 gen.add("near_plane", double_t, 0, "Near plane", 0.01, 0.0, 1.0)
 gen.add("far_plane", double_t, 0, "Far plane", 10.0, 0.0, 10.0)
 
+# Serial identification
+gen.add("use_serial", bool_t, 0, "Enable serial number identification", False)
+
 # Modes
 mode_close = gen.const("close", str_t, "close", "Close mode")
 mode_long = gen.const("long", str_t, "long", "Long mode")

--- a/softkinetic_camera/launch/softkinetic_camera_demo.launch
+++ b/softkinetic_camera/launch/softkinetic_camera_demo.launch
@@ -11,6 +11,7 @@ The arguments given are the device indices of the cameras determined by the Dept
     <param name="rgb_calibration_file" type="string" value="$(find softkinetic_camera)/resources/senz3d.yaml" />
     <param name="confidence_threshold" type="int" value="200" />
     <param name="use_radius_outlier_filter" type="bool" value="false" />
+    <param name="use_serial" type="bool" value="false" />
     <param name="search_radius" type="double" value="0.05" />
     <param name="min_neighbours" type="int" value="50" />
 
@@ -27,5 +28,5 @@ The arguments given are the device indices of the cameras determined by the Dept
 
   <node pkg="rviz" type="rviz" name="softkinect_rviz" respawn="false"  required="true"
 	args="-d $(find softkinetic_camera)/launch/softkinetic.rviz"  />
-  
+
 </launch>

--- a/softkinetic_camera/src/softkinetic_start.cpp
+++ b/softkinetic_camera/src/softkinetic_start.cpp
@@ -169,6 +169,10 @@ DepthSense::CompressionType color_compression;
 DepthSense::FrameFormat color_frame_format;
 int color_frame_rate;
 
+/* Serial Number parameters */
+bool use_serial;
+std::string serial;
+
 DepthSense::DepthNode::CameraMode depthMode(const std::string& depth_mode_str)
 {
     if ( depth_mode_str == "long" )
@@ -1016,6 +1020,33 @@ int main(int argc, char* argv[])
     {
       device_index = atoi(argv[1]);
     }
+    else
+    {
+      nh.param<bool>("use_serial", use_serial, false);
+      if (use_serial)
+      {
+        nh.getParam("serial", serial);
+        bool serialDeviceFound = false;
+        for (int i=0; i < da.size(); i++)
+        {
+          if (!da[i].getSerialNumber().compare(serial))
+          {
+            device_index = i;
+            serialDeviceFound = true;
+          }
+        }
+        if (serialDeviceFound)
+        {
+          ROS_INFO_STREAM("Serial number device found");
+        }
+        else
+        {
+          ROS_ERROR_STREAM("Serial number device not found !!");
+          ROS_ERROR_STREAM("Required Serial Number: " << serial);
+        }
+      }
+    }
+    ROS_INFO_STREAM("Serial Number: " << da[device_index].getSerialNumber());
 
     g_bDeviceFound = true;
 

